src/Components/SingleArticle.jsx

```diff
@@ -1,40 +1,110 @@
-import { useParams } from 'react-router-dom';
-import { useState, useEffect } from 'react';
+import { useParams } from "react-router-dom";
+import { useState, useEffect } from "react";
 
-import { getSingleArticle } from './Api'
-import createdAtConvertor  from './Utils/createdAtConvertor'
+import Comments from "./Comments";
+import { getSingleArticle, patchVote } from "./Api";
+import createdAtConvertor from "./Utils/createdAtConvertor";
 
 const SingleArticle = () => {
-const { article_id } = useParams()
+  const { article_id } = useParams();
 
-const [article, setSingleArticle] = useState()
-const [isLoading, setIsLoading] = useState(true)
+  const [article, setSingleArticle] = useState();
+  const [isLoading, setIsLoading] = useState(true);
+  const [disabledUp, setDisabledUp] = useState(false);
+  const [disabledDown, setDisabledDown] = useState(false);
+  const [error, setError] = useState(null);
 
-useEffect(() => {
+  const [vote, setVote] = useState(0);
+
+  useEffect(() => {
     getSingleArticle(article_id).then((recArticle) => {
-        setSingleArticle(recArticle)
-        setIsLoading(false)
-        
-    })
-}, []);
-
-
-if(isLoading) return <p className='singleArticleText'>Loading ...</p>
-    return ( 
-       
-        <section className='articleWrapper'>
-        <p className='articleBody'>{article.body}</p>
-        <div className='articleDetails'>
-        <img className='singleImg' src={article.article_img_url}></img>
-        <h4 >{article.title} - #{article.article_id}</h4>
-        <p>by {article.author}</p>
-        <p>{createdAtConvertor(article.created_at)}</p>
-        <p>votes:{article.votes}</p>
+      setSingleArticle(recArticle);
+      setIsLoading(false);
+    });
+  }, []);
+
+  useEffect(() => {
+
+      patchVote(article_id, vote)
```

Comment:
This api call should be in the handleClick, _after_ you've optimistically set the state 😄